platform/linux_platform.go

```diff
@@ -569,6 +573,49 @@ func (p linux) SetupEphemeralDiskWithPath(realPath string) error {
 		return bosherr.WrapError(err, "Mounting data partition")
 	}
 
+	if p.options.ScrubEphemeralDisk {
+		agentVersionFilePath := path.Join(p.dirProvider.DataDir(), ".bosh", "agent_version")
+		stemcellVersionFilePath := path.Join(p.dirProvider.BaseDir(), "bosh", "etc", "stemcell_version")
+		stemcellVersion, err := p.fs.ReadFileString(stemcellVersionFilePath)
+		if err != nil {
+			return bosherr.WrapError(err, "Reading stemcell version file")
+		}
+
+		if !p.fs.FileExists(agentVersionFilePath) {
+			err = p.fs.WriteFileString(agentVersionFilePath, stemcellVersion)
+			if err != nil {
+				return bosherr.WrapError(err, "Writting agent version file")
+			}
+		} else {
+			agentVersion, err := p.fs.ReadFileString(agentVersionFilePath)
+			if err != nil {
+				return bosherr.WrapError(err, "Reading agent version file")
+			}
+
+			if agentVersion != stemcellVersion {
+				sysRunPath := path.Join(p.dirProvider.DataDir(), "sys", "run")
+				_, err = p.diskManager.GetMounter().Unmount(sysRunPath)
```

Comment:
Hi @cppforlife, I'm still a little bit confusing with the situation mentioned in last comments, when upgrading stemcell and using the old disk, `/var/vcap/data/sys/run` should have been mounted. Any suggestions? Thank you .